docs/ReferenceResolution.md

```diff
@@ -58,14 +57,33 @@ Steps for adding a new package dependency to an existing project. Let's say I'm
         ```xml
         <ProductDependencies>
           <!-- ... -->
-          </Dependency>
-            <Dependency Name="System.Banana" Version="0.0.1-beta-1">
+          <Dependency Name="System.Banana" Version="0.0.1-beta-1">
             <Uri>https://github.com/dotnet/corefx</Uri>
             <Sha>000000</Sha>
           </Dependency>
           <!-- ... -->
         </ProductDependencies>
         ```
 
-       If you don't know the commit hash of the source code used to produce "0.0.1-beta-1", you can use `000000` as a placeholder for `Sha`
-       as its value is unimportant and will be updated the next time the bot runs.
+        If you don't know the commit hash of the source code used to produce "0.0.1-beta-1", you can use `000000` as a placeholder for `Sha`
+        as its value is unimportant and will be updated the next time the bot runs.
+
+        If the new dependency comes from dotnet/CoreFx or dotnet/core-setup and is not `Microsoft.NetCore.App`, add a
+        `CoherentParentDependency` attribute to the `<Dependency>` element as shown below. This indicates
+        Microsoft.NETCore.App also brings in the System.Banana content and tells `darc` to make the versions coherent.
```

Comment:
@mmitche is this pretty much the correct way to describe things?

Reply:
A couple tweaks:

If the new dependency comes from dotnet/corefx, add a `CoherentParentDependency` attribute to the `<Dependency>` element as shown below. This indicates the dotnet/corefx dependency version should be determined based on the build that was used to produce the Microsoft.NETCore.App.